ovs/lib/storagerouter.py

```diff
@@ -109,20 +109,21 @@ def get_metadata(storagerouter_guid):
                 for storagedriver_partition in disk_partition.storagedrivers:
                     claimed_space += storagedriver_partition.size if storagedriver_partition.size is not None else 0
                     _directory_used_size = 0
-                    try:
-                        output = client.run('du -Pl {0}'.format(storagedriver_partition.path))
-                    except Exception as ex:
-                        logger.error('Failed to get directory usage for {0}. {1}'.format(storagedriver_partition.path, ex))
-                    else:
+                    if client.dir_exists(storagedriver_partition.path):
                         try:
-                            _used_size, _ = output.split('\t')
-                        except ValueError as ve:
-                            logger.error('Unexpected output for directory {0}. "{1}" . {2}'.format(storagedriver_partition.path, output, ve))
+                            output = client.run('du -d 0 {0}'.format(storagedriver_partition.path))
```

Comment:
Don't know whether this should even be in a try - except
First you check if the path exists and if it does i dont see any reason why the 'du' command would even fail
If it would fail, something else and much more severe will be wrong imo